### PR TITLE
Updates to documentation and examples

### PR DESCRIPTION
**Pull request by jorainer**
- Rename some variables (e.g. `queryValues` -> `queryValue`).
- Expand and reorganize documentation and examples.

### CODE REVIEW

File: R/Matched.R [hunk 1]
```diff
@@ -35,27 +35,42 @@
...
 #' - `[` subset the object selecting `query` object elements to keep with
 #'   parameter `i`. The resulting object will contain all the matches
 #'   for the selected query elements. The `target` object will by default be
 #'   returned as-is.
 #'
+#' - `addMatches`: add new matches to an existing object. Parameters
+#'   `queryValue` and `targetValue` allow to define which element(s) in
+#'   `query` and `target` should be considered matching. If `isIndex = TRUE`,
+#'   both `queryValue` and `targetValue` are considered to be integer indices in
+#'   `query` respectively `target` identifying the matching elements.
```
Comment (andreavicini) [inline R/Matched.R:47]: Maybe I would reword it as "are considered to be integer indices identifying the matching elements in `query` and `target` respectively". But it's not really important

File: R/Matched.R [hunk 2]
```diff
@@ -35,27 +35,42 @@
...
-#'   
-#' It is possible to add new matches to a `Matched` or 
-#' `MatchedSummarizedExperiment` object via `addMatches`. With parameters 
-#' `queryValues` and `targetValues` it is possible to specify values of `query` 
-#' and `target` (if they are 1-dimensional) or one of their columns (specified 
-#' with parameters `queryColname` or `targetColname`) and these values are used 
-#' to establish new matches. By using parameter `isIndex` it is possible to 
-#' interpret `queryValues` and `targetValues` as vector of indexes of the 
+#'
+#' It is possible to add new matches to a `Matched` or
```
Comment (andreavicini) [inline R/Matched.R:67]: Maybe, this being explained also above is no longer necessary

**Review by andreavicini — COMMENTED** [review]
Thanks @jorainer for the updates! They look great to me!

File: R/Matched.R (hunk 1 above)
Comment (jorainer) [reply R/Matched.R:47]: good point! I will update.

File: R/Matched.R (hunk 2 above)
Comment (jorainer) [reply R/Matched.R:67]: What do you mean?

Comment (jorainer) [reply R/Matched.R:67]: Ah, now I got it. I forgot to remove this paragraph. Thanks!

**Review by andreavicini — APPROVED** [review]
Everything seems ok to me! Thanks @jorainer !
